core/amber/src/main/scala/edu/uci/ics/texera/web/resource/WorkflowWebsocketResource.scala

```diff
@@ -36,7 +36,16 @@ class WorkflowWebsocketResource extends LazyLogging {
 
   @OnOpen
   def myOnOpen(session: Session, config: EndpointConfig): Unit = {
-    SessionState.setState(session.getId, new SessionState(session))
+    val sessionState = new SessionState(session)
+    SessionState.setState(session.getId, sessionState)
+    val wid = session.getRequestParameterMap.get("wid").get(0).toLong
+    // hack to refresh frontend run button state
+    sessionState.send(WorkflowStateEvent("Uninitialized"))
+    val workflowState =
+      WorkflowService.getOrCreate(WorkflowIdentity(wid))
+    sessionState.subscribe(workflowState)
+    sessionState.send(ClusterStatusUpdateEvent(ClusterListener.numWorkerNodesInCluster))
+    sessionState.send(RegisterWorkflowIdResponse("workflowId registered"))
```

Comment:
do you still need this response??

Reply:
removed the response for both frontend and backend.